authLdap.php

```diff
@@ -263,39 +262,40 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
         }
 
         $uid = authLdap_get_uid($realuid);
-        $role = '';
-
-        // we only need this if either LDAP groups are disabled or
-        // if the WordPress role of the user overrides LDAP groups
-        if (!$authLDAPGroupEnable || !$authLDAPGroupOverUser) {
-            $role = authLdap_user_role($uid);
-        }
+        $roles = authLdap_user_roles($uid);
 
         // do LDAP group mapping if needed
-        // (if LDAP groups override worpress user role, $role is still empty)
-        if (empty($role) && $authLDAPGroupEnable) {
-            $role = authLdap_groupmap($realuid, $dn);
-            authLdap_debug('role from group mapping: ' . $role);
+        if ($authLDAPGroupEnable) {
+            $roles_ldap = authLdap_groupmap($realuid, $dn);
```

Comment:
Well, no, I was getting them separately so I could then either replace $roles with $roles_ldap or merge the two depending on the option selected in the UI.